### ISSUE TITLE
Open Groovy Console on a particular script file in the project

### ISSUE THREAD

**Issue by mray271** [groovy/GMavenPlus#165]
From BASH command line I can do:

```
groovyConsole src/main/scripts/myscript.groovy
```
and have an interactive playground to demo some code.  

The maven snippet in the documentation (below) opens a console, which is great, but is there a way to feed it a script file path so it opens on the intended one? The documentation isn't clear if there is a way to achieve this.

```
       <plugin>
        <groupId>org.codehaus.gmavenplus</groupId>
        <artifactId>gmavenplus-plugin</artifactId>
        <version>1.9.1</version>
        <dependencies>
          <dependency>
            <groupId>org.codehaus.groovy</groupId>
            <artifactId>groovy-all</artifactId>
            <version>3.0.5</version>
            <type>pom</type>
            <scope>runtime</scope>
          </dependency>
        </dependencies>
      </plugin>  
```

**Comment by keeganwitt** [comment]
I just realized I had a miss in the documentation (http://groovy.github.io/GMavenPlus/execute-mojo.html#scripts).  I neglected to mention that you can use local files without using a URL to refer to them.  I'll update this in the next release.

However, https://github.com/groovy/GMavenPlus/wiki/Examples#execute-scripts shows an example of an in-POM script, a script file referred to by URL, and a script file referred to by just the filename.

**Comment by mray271** [comment]
I'm not quite understanding. I want the graphical groovy console to open with a script pre-opened in the editor pane. 

Would I run with

```
mvn gplus:console
```
and in pom.xml I'd specify the script like:

```
       <plugin>
        <groupId>org.codehaus.gmavenplus</groupId>
        <artifactId>gmavenplus-plugin</artifactId>
        <version>1.9.1</version>
        <configuration>
          <scripts>
            <script>${project.basedir}/src/main/scripts/demo_script.groovy</script>
          </scripts>
        </configuration>
        <dependencies>
          <dependency>
            <groupId>org.codehaus.groovy</groupId>
            <artifactId>groovy-all</artifactId>
            <version>3.0.5</version>
            <type>pom</type>
            <scope>runtime</scope>
          </dependency>
        </dependencies>
      </plugin> 
```

This is opening an empty Groovy Console for me.

**Comment by keeganwitt** [comment]
Sorry, I answered a completely different question.  Not sure how that happened.

I don't have any property that would allow that at the moment, but I can look into it.  Was this just a convenience to not have to open the file in the GUI? Or was there some programmatic use case your had in mind?

**Comment by mray271** [comment]
I'm thinking of having an interactive playground to demo some code for people new to a project. 

I guess the idea is like a project-specific version of W3School's TryIt utility

https://www.w3schools.com/js/tryit.asp?filename=tryjson_stringify

I don't want the user to have to start from nothing, so I provide them some relevant code.  There could be serveral of these "tutorial" style demos in a project hooked up the pom.xml.

It would be nice if the different demos could get called out by some short nicknames e.g. 

```
mvn gplus:console helloWorldDemo
```

```
mvn gplus:console advancedDemo
```

And so on...

**Comment by keeganwitt** [comment]
I don't think I can do it in the way you have, since Maven will eat that argument before it can get to the plugin, but we could use a property, like this `mvn gplus:console -DconsoleScript = src/scripts/foo.groovy`.

**Comment by mray271** [comment]
That would definitely work! Thanks  

Still wondering if any way to abstract away the clunky path details if it could be configured with a maven property or similar and then referenced?  This is just an idea, not sure if it's practical.

 e.g.

```
<properties>
     <helloWorldDemo>src/main/scripts/basicThing.groovy</helloWorldDemo>
     <advancedDemo>src/main/scripts/advancedThing.groovy</advancedDemo>
</properties>
```

Then would it be possible to do use those "nicknames"
```
mvn gplus:console -DconsoleScriptProperty=helloWorldDemo
```
```
mvn gplus:console -DconsoleScriptProperty=advancedDemo
```

**Comment by keeganwitt** [comment]
Yea, I suppose I could make it a string and check if the string refers to a file that exists or if the property value exists as a file.

**Comment by mray271** [comment]
Should line [143](https://github.com/groovy/GMavenPlus/blob/5e70d02f5bb9ba199e4f619da2d5df7aa18ddc47/src/main/java/org/codehaus/gmavenplus/mojo/ConsoleMojo.java#L143)
have 3 inputs as in line [139](https://github.com/groovy/GMavenPlus/blob/5e70d02f5bb9ba199e4f619da2d5df7aa18ddc47/src/main/java/org/codehaus/gmavenplus/mojo/ConsoleMojo.java#L139)  ?  I'm thinking missing console as 2nd input is leading to a exception I am seeing when using a property to store the file path of the script.

I'm seeing the exception below:
```
ERROR] Failed to execute goal org.codehaus.gmavenplus:gmavenplus-plugin:1.11.0:console (default-cli) on project foo: Execution default-cli of goal org.codehaus.gmavenplus:gmavenplus-plugin:1.11.0:console failed: object is not an instance of declaring class -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.codehaus.gmavenplus:gmavenplus-plugin:1.11.0:console (default-cli) on project foo: Execution default-cli of goal org.codehaus.gmavenplus:gmavenplus-plugin:1.11.0:console failed: object is not an instance of declaring class
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal org.codehaus.gmavenplus:gmavenplus-plugin:1.11.0:console failed: object is not an instance of declaring class
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:148)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.lang.IllegalArgumentException: object is not an instance of declaring class
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.gmavenplus.util.ReflectionUtils.invokeMethod (ReflectionUtils.java:214)
    at org.codehaus.gmavenplus.mojo.ConsoleMojo.loadScript (ConsoleMojo.java:143)
    at org.codehaus.gmavenplus.mojo.ConsoleMojo.execute (ConsoleMojo.java:108)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

**Comment by mray271** [comment]
If I specify the path directly on the command line as input to -DconsoleScript=<mypath> then it works and the file contents are loaded into the GroovyConsole when it opens.

**Comment by keeganwitt** [comment]
@mray271 Yes, I believe you're right. Sorry about that. I'll correct the typo.

**Comment by keeganwitt** [comment]
This is fixed in 1.11.1.

**Comment by mray271** [comment]
Yes! This works perfectly now! Thanks
